--- a/src/Raft.Core/MessageCodec.cs
+++ b/src/Raft.Core/MessageCodec.cs
@@ -87,7 +87,10 @@
                 return null;
             }
 
-            entries.Add(new LogEntry(et, command));
+            if (et.Value < 0)
+            {
+                entries.Add(new LogEntry(et, command));
+            }
         }
 
         return new AppendEntries(t, leader, pi, pt, entries, commit, round);
