diff --git a/src/Raft.Simulation/Simulator.cs b/src/Raft.Simulation/Simulator.cs
index 8e1cedf..d0f9c50 100644
--- a/src/Raft.Simulation/Simulator.cs
+++ b/src/Raft.Simulation/Simulator.cs
@@ -374,6 +374,7 @@ public sealed class Simulator
                 break;
             default:
                 At(deliverAt, Deliver);
+                At(deliverAt + 1, Deliver);
                 break;
         }
 
