--- a/src/Raft.Core/EntryLog.cs
+++ b/src/Raft.Core/EntryLog.cs
@@ -391,7 +391,7 @@
         }
 
         var at = (int)(index - BaseIndex) - 1;
-        _fileLength = at == 0 ? Snapshot?.EndOffset ?? 0 : _entries[at - 1].EndOffset;
+        _fileLength = _entries[at].EndOffset - 32 - _entries[at].Command.Length;
         _entries.RemoveRange(at, _entries.Count - at);
         while (_configurations.Count > 0 && _configurations[^1] >= index)
         {
